### PR TITLE
feature: add support to keda scaledobject v1alpha1

### PR DESCRIPTION
**Pull request by jonathanmdr**
I believe it provides support to pdb-min-available when the Replica number is controlled via Keda ScaledObject

I closed erroneously this PR: https://github.com/stackrox/kube-linter/pull/703

### CODE REVIEW

File: pkg/extract/hpa_spec.go [hunk 1]
```diff
@@ -43,6 +46,8 @@ func HPAScaleTargetRefName(obj k8sutil.Object) (string, bool) {
 		return hpa.Spec.ScaleTargetRef.Name, true
 	case *autoscalingV1.HorizontalPodAutoscaler:
 		return hpa.Spec.ScaleTargetRef.Name, true
+	case *kedaV1Alpha1.ScaledObject:
+		return hpa.Spec.ScaleTargetRef.Name, true
```
Comment (janisz) [inline pkg/extract/hpa_spec.go:50]: I this line covered by a unit test?

**Review by janisz — CHANGES_REQUESTED** [review]
Nice work! On the first look everything is fine but on second there is something missing. 
1. You correctly added keda to `extract` but it's never called in unit test
2. Could you add good/bad tests to verify it's really working
3. You need to register keda scheme in https://github.com/stackrox/kube-linter/blob/e0225f476a85c4533eab934105c63f44e9d95a50/pkg/lintcontext/parse_yaml.go#L47

No related to this PR but we need to use `--fail-on-invalid-resource` in e2e as in your case it will highlight the error

```
pdb-min-available.yaml: (object: <no namespace>/ /, Kind=) failed to decode: no kind "ScaledObject" is registered for version "keda.sh/v1alpha1" in scheme "pkg/runtime/scheme.go:100" (check: failed-to-load-object, remediation: Confirm that the file is accessible and is valid k8s yaml.)
```

File: pkg/extract/hpa_spec.go (hunk 1 above)
Comment (jonathanmdr) [reply pkg/extract/hpa_spec.go:50]: I adjusted the test code to cover this!

**Comment by jonathanmdr** [comment]
> Nice work! On the first look everything is fine but on second there is something missing.
> 
> 1. You correctly added keda to `extract` but it's never called in unit test
> 2. Could you add good/bad tests to verify it's really working
> 3. You need to register keda scheme in https://github.com/stackrox/kube-linter/blob/e0225f476a85c4533eab934105c63f44e9d95a50/pkg/lintcontext/parse_yaml.go#L47
> 
> No related to this PR but we need to use `--fail-on-invalid-resource` in e2e as in your case it will highlight the error
> 
> ```
> pdb-min-available.yaml: (object: <no namespace>/ /, Kind=) failed to decode: no kind "ScaledObject" is registered for version "keda.sh/v1alpha1" in scheme "pkg/runtime/scheme.go:100" (check: failed-to-load-object, remediation: Confirm that the file is accessible and is valid k8s yaml.)
> ```

Well noted!

I adjusted my implementation and unit tests writing good/bad scenarios for this functionality!

If there's any other feedback, we're happy to follow up on it 😄

**Review by janisz — APPROVED** [review]

**Comment by janisz** [comment]
Nice, thank you. Could you rebase?

**Comment by jonathanmdr** [comment]
> Nice, thank you. Could you rebase?

Sure, rebase applied!

**Comment by janisz** [comment]
@jonathanmdr  Thanks! In the meantime dependabot updated k8s so I updated your PR.

**Pull request merged**